File: src/lessons/bat/string1/FrontTimesEntity.java
package lessons.bat.string1;

import static plm.core.ValueSerializer.*;

import plm.universe.bat.BatEntity;

public class FrontTimesEntity extends BatEntity {

  public void run()
  {
    int count = getTestCount();
    for (int i = 0; i < count; i++) {
      Object[] param = (Object[])deserialize(getTest(i));
      setTestResult(i, serialize(frontTimes((String)param[0], (Integer)param[1])));
    }
  }

  /* BEGIN TEMPLATE */
  String frontTimes(String str, int n)
  {
    /* BEGIN SOLUTION */
    int frontLen = 3;
    if (frontLen > str.length()) {
      frontLen = str.length();
    }
    String front = str.substring(0, frontLen);

    String result = "";
    for (int i = 0; i < n; i++) {
      result = result + front;
    }
    return result;
    /* END SOLUTION */
  }
  /* END TEMPLATE */
}

File: src/lessons/bat/string1/FrontTimesEntity.scala
package lessons.bat.string1

import plm.core.ValueSerializer
import plm.universe.bat.BatEntity

class FrontTimesEntity extends BatEntity {

    override def run() {
      import ValueSerializer._

      val count = getTestCount()
      for (i <- 0 to count -1) {
        val param = deserialize(getTest(i)).asInstanceOf[Array[Object]]
        setTestResult(i, serialize(frontTimes(param(0).asInstanceOf[String], param(1).asInstanceOf[Int])) )
	  }
	}

	/* BEGIN TEMPLATE */
	def frontTimes(str:String, n:Int):String = {
	/* BEGIN SOLUTION */
	  var frontLen = 3
	  if (frontLen > str.length)
	    frontLen = str.length
	  var front = ""
	  if (str.length >0)
	    front = str.substring(0,frontLen)
	  return front * n
	/* END SOLUTION */
	}
	/* END TEMPLATE */
}
